### PR TITLE
Add ability to copy data from pda

### PR DESCRIPTION
**Pull request by user424242420**
Add ability to copy station name and time to clipboard
## About the PR

I change form of pda in xmarin. Add two buttons to copy needable data/
## Why / Balance

Because that can speed up documen flow
## Technical details

I just add two buttons add connect robust interface to copy text.
## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Nothing special
**Changelog**

:cl: username
- add: Ability to copy data from pda

### CODE REVIEW

File: Content.Client/PDA/PdaMenu.xaml.cs [hunk 1]
```diff
@@ -84,6 +87,17 @@ public PdaMenu()
                 ToHomeScreen();
             };
 
+            CopyStationNameButton.OnPressed += _ =>
+            {
+                _clipboard.SetText(_StationName?? Loc.GetString("comp-pda-ui-unknown"));
```
Comment (deltanedas) [inline Content.Client/PDA/PdaMenu.xaml.cs:92]: _StationName should be renamed to _stationName, and its not optional so dont need the ?? bit

File: Content.Client/PDA/PdaMenu.xaml.cs [hunk 2]
```diff
@@ -84,6 +87,17 @@ public PdaMenu()
...
+            };
+
+            CopyStationTimeButton.OnPressed += _ =>
+            {
+                var stationTime = _gameTiming.CurTime.Subtract(_gameTicker.RoundStartTimeSpan);
+                _clipboard.SetText((stationTime.ToString("hh\\:mm\\:ss") ?? Loc.GetString("comp-pda-ui-unknown")));
```
Comment (deltanedas) [inline Content.Client/PDA/PdaMenu.xaml.cs:98]: ToString wont return null

File: Content.Client/PDA/PdaMenu.xaml.cs [hunk 3]
```diff
@@ -113,6 +127,7 @@ public void UpdateState(PdaUpdateState state)
 
             StationNameLabel.SetMarkup(Loc.GetString("comp-pda-ui-station",
                 ("station", state.StationName ?? Loc.GetString("comp-pda-ui-unknown"))));
+            _StationName = state.StationName ?? Loc.GetString("comp-pda-ui-unknown");
```
Comment (deltanedas) [inline Content.Client/PDA/PdaMenu.xaml.cs:130]: move it a line up and reuse

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by user424242420** [comment]
Yes. I fix that you said, and add ability to copy name and profession.

**Review by ficcialfaint — DISMISSED** [review]
The buttons just take extra space. You can just make the text selectable or make it copy on clicking on the text or you can use your own way of simplifying it

**Comment by chromiumboy** [comment]
I'm not sure what purpose this PR fulfills. Adding extra buttons to the PDA screen is just unnecessary UI clutter

**Comment by user424242420** [comment]
> I'm not sure what purpose this PR fulfills. Adding extra buttons to the PDA screen is just unnecessary UI clutter

I got it, i'll make small copy button instead of current pr.

**Comment by user424242420** [comment]
> I'm not sure what purpose this PR fulfills. Adding extra buttons to the PDA screen is just unnecessary UI clutter

But this buttons needs for documents flow on rp servers. I tired to write information from pda again, and again.

**Comment by chromiumboy** [comment]
> > I'm not sure what purpose this PR fulfills. Adding extra buttons to the PDA screen is just unnecessary UI clutter
> 
> But this buttons needs for documents flow on rp servers. I tired to write information from pda again, and again.

As faint said, you should either make the text selectable, or auto-copy on clicking the text. These buttons are not the way to do it

**Comment by user424242420** [comment]
But how people will get, that they need to press on text to copt in buffer? Not all people reads changelog

> > > I'm not sure what purpose this PR fulfills. Adding extra buttons to the PDA screen is just unnecessary UI clutter
> > 
> > 
> > But this buttons needs for documents flow on rp servers. I tired to write information from pda again, and again.
> 
> As faint said, you should either make the text selectable, or auto-copy on clicking the text. These buttons are not the way to do it

**Comment by user424242420** [comment]
I will make as you said, but problem the same: how people will understand this feature

**Comment by chromiumboy** [comment]
If someone wants to copy text, the first thing they'll try is to click on it, so I'm not that worried about discoverability. In any case, the use case is so niche that its not going an issue for the vast majority of players

**Comment by user424242420** [comment]
@ficcialfaint i change as you said, pls check it

File: Content.Client/PDA/PdaMenu.xaml [hunk 4]
```diff
@@ -28,20 +28,24 @@
...
+            <ContainerButton Name = "IdInfoButton" Access="Public">
+                <RichTextLabel Name="IdInfoLabel" Access="Public"/>
+            </ContainerButton>
+            <ContainerButton Name = "StationNameButton" Access="Public">
+                <RichTextLabel Name="StationNameLabel" Access="Public"/>
+            </ContainerButton>
+            <ContainerButton Name="StationAlertLevelButton" Access="Public">
+                <RichTextLabel Name="StationAlertLevelLabel" Access="Public" />
+            </ContainerButton>
+            <ContainerButton Name = "StationTimeButton" Access="Public">
```
Comment (deathride58) [inline Content.Client/PDA/PdaMenu.xaml:43]: The extraneous spaces here should probably be removed for cleaner XAML

```suggestion
            <ContainerButton Name="PdaOwnerButton" Access="Public">
                <RichTextLabel Name="PdaOwnerLabel" Access="Public" />
            </ContainerButton>
            <ContainerButton Name="IdInfoButton" Access="Public">
                <RichTextLabel Name="IdInfoLabel" Access="Public"/>
            </ContainerButton>
            <ContainerButton Name="StationNameButton" Access="Public">
                <RichTextLabel Name="StationNameLabel" Access="Public"/>
            </ContainerButton>
            <ContainerButton Name="StationAlertLevelButton" Access="Public">
                <RichTextLabel Name="StationAlertLevelLabel" Access="Public" />
            </ContainerButton>
            <ContainerButton Name="StationTimeButton" Access="Public">
```

**Review by deathride58 — CHANGES_REQUESTED** [review]
Just a minute codesmell thing, but the rest of the PR does look good

File: Content.Client/PDA/PdaMenu.xaml (hunk 4 above)
Comment (ficcialfaint) [reply Content.Client/PDA/PdaMenu.xaml:43]: Also public access for the button seems to be useless here(?)

Comment (user424242420) [reply Content.Client/PDA/PdaMenu.xaml]: Fix public access, fix spaces beetwen ids

**Comment by user424242420** [comment]
@deathride58 @ficcialfaint All your requests have done. Please check it.

**Review by deathride58 — APPROVED** [review]

**Comment by juliangiebel** [comment]
Can you provide a Video of what clicking on the text looks like? And where does the error sprite come from?

**Comment by user424242420** [comment]
> Can you provide a Video of what clicking on the text looks like? And where does the error sprite come from? The code looks good though

@juliangiebel 

https://github.com/space-wizards/space-station-14/assets/142989209/2cd86d0f-307e-4729-bc98-5520921a06cb

Error sprite was your problem. It was fixed some where.

**Comment by MACMAN2003** [comment]
would it be possible for the text to become slightly darker upon hovering over it with the mouse?

**Pull request merged**